plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerTableBase.java

```diff
@@ -248,10 +248,12 @@ private void enableIdentityInsert(DBCSession session, boolean enable) throws DBC
         }
     }
 
-    private boolean hasIdentInsert(@NotNull DBSManipulationType type, @NotNull DBSAttributeBase[] attributes) {
+    private boolean hasIdentityInsert(@NotNull DBSManipulationType type, @NotNull DBSAttributeBase[] attributes) {
         if (type == DBSManipulationType.INSERT) {
             for (DBSAttributeBase attr : attributes) {
-                if (attr.isAutoGenerated()) {
+                if (attr instanceof SQLServerTableColumn && ((SQLServerTableColumn) attr).isIdentity()) {
+                    return true;
+                } else if (attr.isAutoGenerated()) {
```

Comment:
what about

if ((attr instanceof SQLServerTableColumn && ((SQLServerTableColumn) attr).isIdentity()) || attr.isAutoGenerated())

?

Reply:
Looks messy, attempted to make it eye-friendly.